components/docker/docker.go

```diff
@@ -0,0 +1,170 @@
+package docker
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/common/namer"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/DataDog/test-infra-definitions/components/command"
+	remoteComp "github.com/DataDog/test-infra-definitions/components/remote"
+
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	composeVersion = "v2.12.2"
+	defaultTimeout = 300
+)
+
+type Manager struct {
+	namer namer.Namer
+	host  *remoteComp.Host
+
+	opts []pulumi.ResourceOption
+}
+
+func NewManager(e config.CommonEnvironment, host *remoteComp.Host, installDocker bool, opts ...pulumi.ResourceOption) (*Manager, pulumi.Resource, error) {
+	manager := &Manager{
+		namer: e.CommonNamer.WithPrefix("docker"),
+		host:  host,
+		opts:  opts,
+	}
+
+	var err error
+	var installCmd pulumi.Resource
+	if installDocker {
+		installCmd, err = manager.install()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		manager.opts = utils.MergeOptions(manager.opts, utils.PulumiDependsOn(installCmd))
+	}
+
+	composeCmd, err := manager.installCompose()
+	if err != nil {
+		return nil, nil, err
+	}
+	manager.opts = utils.MergeOptions(manager.opts, utils.PulumiDependsOn(composeCmd))
+
+	return manager, nil, nil
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
I think the second one used to be the `installCmd`. If we always return nil then we can remove it from the return type

Reply:
It should be `composeCmd` indeed, fixed.